extensions/cms/src/cmsUtils.ts

```diff
@@ -143,63 +185,110 @@ export class CmsUtils {
+		if (connection && connection.options) {
+			if (connection.options.server === parentServerName) {
+				// error out for same server registration
+				let errorText = localize('cms.errors.sameServerUnderCms', 'You cannot add a shared registered server with the same name as the Configuration Server');
+				this.showErrorMessage(errorText);
+				throw new Error(errorText);
+			} else {
+				let registeredServerName = connection.options.registeredServerName === '' ? connection.options.server : connection.options.registeredServerName;
+				let result = await provider.addRegisteredServer(ownerUri, relativePath, registeredServerName, connection.options.registeredServerDescription, connection);
+				return result;
 			}
-		});
+		}
 	}
 
 	public async removeRegisteredServer(registeredServerName: string, relativePath: string, ownerUri: string): Promise<boolean> {
 		let provider = await this.getCmsService();
-		return provider.removeRegisteredServer(ownerUri, relativePath, registeredServerName).then((result) => {
-			return result;
-		});
+		let result = await provider.removeRegisteredServer(ownerUri, relativePath, registeredServerName);
+		return result;
 	}
 
 	public async addServerGroup(groupName: string, groupDescription: string, relativePath: string, ownerUri: string): Promise<boolean> {
 		let provider = await this.getCmsService();
-		return provider.addServerGroup(ownerUri, relativePath, groupName, groupDescription).then((result) => {
-			return result;
-		});
+		let result = await provider.addServerGroup(ownerUri, relativePath, groupName, groupDescription);
+		return result;
 	}
 
 	public async removeServerGroup(groupName: string, relativePath: string, ownerUri: string): Promise<boolean> {
 		let provider = await this.getCmsService();
-		return provider.removeServerGroup(ownerUri, relativePath, groupName).then((result) => {
-			return result;
-		});
+		let result = await provider.removeServerGroup(ownerUri, relativePath, groupName);
+		return result;
 	}
 
 	// Getters
 	public get registeredCmsServers(): ICmsResourceNodeInfo[] {
 		return this._registeredCmsServers;
 	}
 
-	public get connection(): Thenable<azdata.connection.Connection> {
-		return this.openConnectionDialog([cmsProvider], undefined, { saveConnection: false }).then((connection) => {
-			if (connection) {
-				// remove group ID from connection if a user chose connection
-				// from the recent connections list
-				connection.options['groupId'] = null;
-				connection.providerName = mssqlProvider;
-				return connection;
+	public async credentialProvider(): Promise<azdata.CredentialProvider> {
+		if (!this._credentialProvider) {
+			this._credentialProvider = await azdata.credentials.getProvider(CredentialNamespace);
+		}
+		return this._credentialProvider;
+	}
+
+	public async connection(initialConnectionProfile?: azdata.IConnectionProfile): Promise<azdata.connection.Connection> {
```

Comment:
I think generally method names should contain a verb and optionally a noun if needed.  so `connect` or `establishConnection` would be better names.